hazelcast/src/main/java/com/hazelcast/map/impl/LocalMapStatsProvider.java

```diff
@@ -354,32 +354,33 @@ private void addIndexStats(String mapName, LocalMapStatsImpl localMapStats) {
         if (mapContainer == null) {
             return;
         }
-        Indexes globalIndexes = mapContainer.getIndexes();
-
+        IndexRegistry globalIndexRegistry = mapContainer.getGlobalIndexRegistry();
         Map<String, OnDemandIndexStats> freshStats = null;
-        if (globalIndexes != null) {
-            assert globalIndexes.isGlobal();
-            localMapStats.setQueryCount(globalIndexes.getIndexesStats().getQueryCount());
-            localMapStats.setIndexedQueryCount(globalIndexes.getIndexesStats().getIndexedQueryCount());
-            freshStats = aggregateFreshIndexStats(globalIndexes.getIndexes(), null);
+        if (globalIndexRegistry != null) {
+            assert globalIndexRegistry.isGlobal();
+            localMapStats.setQueryCount(globalIndexRegistry.getIndexesStats().getQueryCount());
+            localMapStats.setIndexedQueryCount(globalIndexRegistry.getIndexesStats().getIndexedQueryCount());
+            freshStats = aggregateFreshIndexStats(globalIndexRegistry.getIndexes(), null);
             finalizeFreshIndexStats(freshStats);
         } else {
             long queryCount = 0;
             long indexedQueryCount = 0;
-            PartitionContainer[] partitionContainers = mapServiceContext.getPartitionContainers();
-            for (int i = 0; i < partitionContainers.length; i++) {
-                PartitionContainer partitionContainer = partitionContainers[i];
-                IPartition partition = partitionService.getPartition(partitionContainer.getPartitionId());
+
+            int partitionCount = partitionService.getPartitionCount();
+            for (int partitionId = 0; partitionId < partitionCount; partitionId++) {
+                IPartition partition = partitionService.getPartition(partitionId);
                 if (!partition.isLocal()) {
                     continue;
                 }
 
-                Indexes partitionIndexes = partitionContainer.getIndexes().get(mapName);
-                if (partitionIndexes == null) {
+                IndexRegistry partitionedIndexRegistry
+                        = mapContainer.getOrNullPartitionedIndexRegistry(partitionId);
+                if (partitionedIndexRegistry == null) {
                     continue;
                 }
-                assert !partitionIndexes.isGlobal();
-                IndexesStats indexesStats = partitionIndexes.getIndexesStats();
+
+                assert !partitionedIndexRegistry.isGlobal();
```

Comment:
If we split global vs local registry, why do we need isGlobal() method?

Reply:
no new change i introduced here also, as far as i see, isGlobal() is used to introduce different behavior or making assertions.